plugins/devtools-backend/src/api/DevToolsBackendApi.ts

```diff
@@ -0,0 +1,222 @@
+      return result;
+    }
+    for (const endpoint of endpoints) {
+      this.logger?.info(
+        `Checking external dependency "${endpoint.name}" at "${endpoint.target}"`,
+      );
+
+      switch (endpoint.type) {
+        case 'ping': {
+          const pingResult = await this.pingExternalDependency(endpoint);
+          result.push(pingResult);
+          break;
+        }
+        case 'fetch': {
+          const fetchResult = await this.fetchExternalDependency(endpoint);
+          result.push(fetchResult);
+          break;
+        }
+        default:
+          return result;
+      }
+    }
+
+    return result;
+  }
+
+  private async fetchExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    let status;
+    let error;
+
+    await fetch(endpoint.target)
+      .then(res => {
+        status = res.status === 200 ? 'Healthy' : 'Unhealthy';
+        this.logger.info(
+          `Fetch for ${endpoint.name} resulted in status code "${res.status}"`,
+        );
+      })
+      .catch((err: Error) => {
+        this.logger.error(`Fetch failed for ${endpoint.name} - ${err.message}`);
+        error = err.message;
+      });
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: status ?? 'Unhealthy',
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  private async pingExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    const pingResult = await ping.promise.probe(endpoint.target);
+
+    let error;
+    if (
+      pingResult.packetLoss === '100.000' ||
+      pingResult.packetLoss === 'unknown'
+    ) {
+      this.logger.error(
+        `Ping failed for ${endpoint.name} - ${pingResult.output}`,
+      );
+      error = pingResult.output === '' ? `${endpoint.target} - Unknown` : pingResult.output;
+    }
+
+    this.logger.info(`Ping results for ${endpoint.name}: ${pingResult.output}`);
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: pingResult.alive ? 'Healthy' : 'Unhealthy',
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  public async listConfig(): Promise<JsonValue> {
+    const args = parseArgs(process.argv);
+
+    const configTargets: ConfigTarget[] = [args.config ?? []]
+      .flat()
+      .map(arg =>
+        isValidUrl(arg) ? { url: arg } : { path: resolvePath(arg) },
+      );
+
+    /* eslint-disable-next-line no-restricted-syntax */
+    const paths = findPaths(__dirname);
+
+    const { packages } = await getPackages(paths.targetDir);
+    const schema = await loadConfigSchema({
+      dependencies: packages.map(p => p.packageJson.name),
+    });
+
+    const { appConfigs } = await loadConfig({
+      configRoot: paths.targetRoot,
+      configTargets: configTargets,
+    });
```

Comment:
No reason to load in config again here I think? Best to depend on the config that's available from the backend env.

Reply:
Alright, I agree, but I've had a hard time making this happen. The config I have is of type `Config` but when I want to say filter the secrets I need it to be a type `AppConfig`. these are both very different and I'm not sure how to go from one to the other.

I could use some suggestions 🤔

Reply:
I think you could do something like this,
```
    const config = { data: this.config.get() as JsonObject, context: 'inline' };

    const sanitizedConfigs = schema.process([config], {
      // Needed this to handle the config.get() removing keys that map to empty env vars.
      ignoreSchemaErrors: true,
      valueTransform: (value, context) => 
         context.visibility === 'secret' ? '<secret>' : value,
    });
```
The value returned by `this.config.get()` will contain all resolved keys so we don't need to pull the config a second time.

Semi-related note, it looks like there is no config schema for the `auth-backend` plugin which means both client secrets and SAML certs would be leaked with this plugin.

Reply:
I've mostly implemented @sennyeya suggestion where things are different is that I'm setting `ignoreSchemaErrors` to `false`. If there are any errors I grab those and then generate the config with `ignoreSchemaErrors` to `true`. This way people will be able to see why things might not be working like environment variables not being set.